pkg/proxy/ipvs/proxier.go

```diff
@@ -767,30 +759,18 @@ func cleanupIptablesLeftovers(ipt utiliptables.Interface) (encounteredError bool
 	}
 
 	// Flush and remove all of our chains.
-	iptablesData := bytes.NewBuffer(nil)
-	if err := ipt.SaveInto(utiliptables.TableNAT, iptablesData); err != nil {
-		glog.Errorf("Failed to execute iptables-save for %s: %v", utiliptables.TableNAT, err)
-		encounteredError = true
-	} else {
-		existingNATChains := utiliptables.GetChainLines(utiliptables.TableNAT, iptablesData.Bytes())
-		natChains := bytes.NewBuffer(nil)
-		natRules := bytes.NewBuffer(nil)
-		writeLine(natChains, "*nat")
-		// Start with chains we know we need to remove.
-		for _, chain := range []utiliptables.Chain{kubeServicesChain, kubePostroutingChain, KubeMarkMasqChain, KubeServiceIPSetsChain} {
-			if _, found := existingNATChains[chain]; found {
-				chainString := string(chain)
-				writeLine(natChains, existingNATChains[chain]) // flush
-				writeLine(natRules, "-X", chainString)         // delete
+	for _, chain := range []utiliptables.Chain{kubeServicesChain, kubePostroutingChain, KubeMarkMasqChain} {
```

Comment:
This should probably not touch MARK-MASQ - that rule is shared with kubelet, too.  Unfortunately.

Reply:
I found kubelet will ensure MARK-MASQ by itself every 1 min, see https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_network.go#L206

Reply:
IIUC, kubelet will not clean KUBE-MARK-MASQ chain. Then, KUBE-MARK-MASQ will be left forever if kube-proxy doesn't clean it?